publish_trips.py

```diff
@@ -113,12 +117,42 @@ def map_row(row):
         if key in FIELDS
     }
 
+def classify_bike_type(data):
+    """
+    Classifying Bikes into Electric or Classic types based on the ID pattern supplied by CapMetro.
+    All e-bikes will have 5 digit bike numbers starting with “15” and up.
+    """
+    for row in data:
+        if len(row["bicycle_id"]) == 5 and int(row["bicycle_id"][0:2]) >= 15:
+            row["bike_type"] = "Electric Bike"
+        # Additional case where the ID is a six character ID with a trailing E are also E-bikes
+        elif len(row["bicycle_id"]) == 6 and int(row["bicycle_id"][0:2]) >= 15 and row["bicycle_id"][5] == "E":
+            row["bike_type"] = "Electric Bike"
+        else:
+            row["bike_type"] = "Classic Bike"
+
+    return data
+
+def populate_month_year(data):
+    """
+    Extracts the month and year of the checkout date for two columns in the Socrata dataset
+    """
+    for row in data:
+        date = dateutil.parser.parse(row["checkout_date"])
+        row["year"] = date.strftime("%Y")
+        row["month"] = date.strftime("%-m")
+    return data
```

Comment:
We have two columns in Socrata `month` and `year` that are mostly empty so it would be good for us to repopulate them when we re-process the data.